### PR TITLE
Virtualize all NetIDs to reduce network traffic

### PR DESCRIPTION
**Pull request by Tyler-IN**
The serializer encodes the runtime type, so we can take advantage of it to reduce network usage

### CODE REVIEW

**Comment by Tyler-IN** [comment]
Linking https://github.com/space-wizards/RobustToolbox/pull/1116

**Comment by Tyler-IN** [comment]
All relevant changes except for `Material.cs`; I was looking for regex replace artifacts by looking for `^public` and saw it.

File: Content.Shared/GameObjects/Components/Chemistry/SharedSolutionComponent.cs [hunk 1]
```diff
@@ -20,7 +20,9 @@ public class SharedSolutionComponent : Component
         [Serializable, NetSerializable]
         public class SolutionComponentState : ComponentState
         {
-            public SolutionComponentState() : base(ContentNetIDs.SOLUTION) { }
+            public override uint NetID => ContentNetIDs.SOLUTION;
+
+        public SolutionComponentState() { }
```
Comment (ComicIronic) [inline Content.Shared/GameObjects/Components/Chemistry/SharedSolutionComponent.cs:25]: Funky indent.

Comment (Tyler-IN) [reply Content.Shared/GameObjects/Components/Chemistry/SharedSolutionComponent.cs:25]: Good catch.

**Pull request merged**